aiohttp/connector.py

```diff
@@ -1117,6 +1117,9 @@ def drop_exception(fut: "asyncio.Future[List[Dict[str, Any]]]") -> None:
             port = hinfo["port"]
 
             try:
+                # Strip trailing dots, certificates contain FQDN without dots.
+                # See https://github.com/aio-libs/aiohttp/issues/3636
+                server_hostname = hinfo["hostname"].rstrip(".") if sslcontext else None
```

Comment:
I just realized this violates the rule of having only one instruction in a try-block. Let's fix this:
```suggestion
            # Strip trailing dots, certificates contain FQDN without dots.
            # See https://github.com/aio-libs/aiohttp/issues/3636
            server_hostname = hinfo["hostname"].rstrip(".") if sslcontext else None
            try:
```